### PR TITLE
Add AudioEngine js bindings

### PR DESCRIPTION
**Pull request by pandamicro**

### CODE REVIEW

**Comment by pandamicro** [comment]
@WenhaiLin Please review

**Comment by WenhaiLin** [comment]
@pandamicro You miss to update win32/window-universal/window-10 projects.

File: tests/cpp-tests/Classes/NewAudioEngineTest/NewAudioEngineTest.cpp [hunk 1]
```diff
@@ -169,94 +160,19 @@ namespace {
...
-        
-        virtual void onTouchCancelled(Touch *touch, Event *unusedEvent) override{
-            _touchEvent = TouchEvent::CANCEL;
-            Slider::onTouchCancelled(touch, unusedEvent);
-            
-            if(_callback){
-                _callback(this,_ratio,_touchEvent);
-            }
+        float getRatio () {
+            return _ratio;
```
Comment (WenhaiLin) [inline tests/cpp-tests/Classes/NewAudioEngineTest/NewAudioEngineTest.cpp:172]: You should update the value of `_ratio`.
eg:

``` cpp
    _ratio = 1.f * _percent / _maxPercent;
```

File: tests/cpp-tests/Classes/NewAudioEngineTest/NewAudioEngineTest.cpp [hunk 2]
```diff
@@ -376,17 +293,21 @@ bool AudioControlTest::init()
...
                 _updateTimeSlider = false;
                 break;
-            case SliderEx::TouchEvent::UP:
+            case Slider::EventType::ON_SLIDEBALL_UP:
                 if (_audioID != AudioEngine::INVALID_AUDIO_ID && _duration != AudioEngine::TIME_UNKNOWN) {
-                    AudioEngine::setCurrentTime(_audioID,_duration * ratio);
+                    float ratio = (float)slider->getPercent() / 100;
+                    ratio = clampf(ratio, 0.0f, 1.0f);
+                    AudioEngine::setCurrentTime(_audioID, _duration * ratio);
+                    slider->setRatio(ratio);
```
Comment (WenhaiLin) [inline tests/cpp-tests/Classes/NewAudioEngineTest/NewAudioEngineTest.cpp]: This line of code is unnecessary.

File: tests/js-tests/src/NativeTest/AudioEngineTest.js [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+                    this._timeSlider.setRatio(this._timeRatio);
+                }
+            }
+        }
+    }
+});
+
+window.AudioTestFlow = new FlowControl([
+    AudioControlTest,
+    // PlaySimultaneouslyTest,
```
Comment (WenhaiLin) [inline tests/js-tests/src/NativeTest/AudioEngineTest.js]: //todo?

File: tests/js-tests/src/NativeTest/AudioEngineTest.js [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+    // AudioProfileTest,
+    // InvalidAudioFileTest,
+    // LargeAudioFileTest
+]);
+
+})(window);
```
Comment (WenhaiLin) [inline tests/js-tests/src/NativeTest/AudioEngineTest.js]: It's better to add a blank line at end of file.

**Comment by WenhaiLin** [comment]
review done

**Comment by pandamicro** [comment]
> @pandamicro You miss to update win32/window-universal/window-10 projects.

JSB doesn't support win10 yet

**Comment by pandamicro** [comment]
[ci rebuild] windows linker error

**Comment by pandamicro** [comment]
Fixed https://github.com/cocos2d/cocos2d-x/pull/12517

**Pull request merged**